File: Jessica/118.Pascal's Triangle.py
class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        if numRows == 0:
            return []
        if numRows == 1:
            return [[1]]
        else:
            outter=[[1]]
            for i in range(1,numRows):
                inner=[1]
                for j in range(1,i):
                    inner.append(outter[i-1][j-1]+outter[i-1][j])
                inner.append(1)
                outter.append(inner)    
        return outter
        

File: Jessica/118.Pascal's Triangle.cpp
class Solution {
public:
    vector<vector<int>> generate(int numRows) {
        
        vector<vector<int>> res;
        for(int i=1;i<=numRows;i++){
            if(i==1 || i==2) {vector<int> temp(i,1);res.push_back(temp);continue;}
            vector<int> temp(i,1);
            for(int j=1;j<i-1;j++){
                temp[j] = res[i-2][j-1] + res[i-2][j];    
            }
            res.push_back(temp);
        }
        return res;
    }
    
};
